bundles/persistence/org.openhab.persistence.h2sql/java/org/openhab/persistence/h2sql/internal/H2SqlPersistenceService.java

```diff
@@ -0,0 +1,464 @@
+ * ContactItem       OnOffType    CHAR(6)
+ * DateTimeItem      DateTimeType DATETIME
+ * DimmerItem        PercentType  TINYINT
+ * NumberItem        DecimalType  DOUBLE
+ * RollershutterItem PercentType  TINYINT
+ * StringItem        StringType   VARCHAR(20000)
+ * SwitchItem        OnOffType    CHAR(3)
+ * 
+ * In the store method, type conversion is performed where the default type for
+ * an item is not as above For example, DimmerType can return OnOffType, so to
+ * keep the best resolution, we store as a number in SQL and convert to
+ * DecimalType before persisting to H2SQL.
+ * 
+ * @author Chris Jackson - Initial contribution
+ * @since 1.8.0
+ */
+public class H2SqlPersistenceService implements QueryablePersistenceService, ManagedService {
+
+    private static final Logger logger = LoggerFactory.getLogger(H2SqlPersistenceService.class);
+
+    private String driverClass = "org.h2.Driver";
+
+    protected ItemRegistry itemRegistry;
+
+    private Connection connection = null;
+
+    private Map<String, String> sqlTypes = new HashMap<String, String>();
+
+    public void activate() {
+        logger.debug("H2SQL: Persistence bundle activate.");
+
+        // Initialise the type array
+        sqlTypes.put("COLORITEM", "VARCHAR(70)");
+        sqlTypes.put("CONTACTITEM", "VARCHAR(6)");
+        sqlTypes.put("DATETIMEITEM", "DATETIME");
+        sqlTypes.put("DIMMERITEM", "TINYINT");
+        sqlTypes.put("GROUPITEM", "DOUBLE");
+        sqlTypes.put("NUMBERITEM", "DOUBLE");
+        sqlTypes.put("ROLERSHUTTERITEM", "TINYINT");
+        sqlTypes.put("STRINGITEM", "VARCHAR(20000)");
+        sqlTypes.put("SWITCHITEM", "CHAR(3)");
+    }
+
+    public void deactivate() {
+        logger.debug("H2SQL: Persistence bundle deactivate. Disconnecting from database.");
+        disconnectFromDatabase();
+    }
+
+    public void setItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = itemRegistry;
+    }
+
+    public void unsetItemRegistry(ItemRegistry itemRegistry) {
+        this.itemRegistry = null;
+    }
+
+    // public void setPersistentStateRestorer(PersistentStateRestorer persistentStateRestorer) {
+    // this.persistentStateRestorer = persistentStateRestorer;
+    // }
+
+    // public void unsetPersistentStateRestorer(PersistentStateRestorer persistentStateRestorer) {
+    // this.persistentStateRestorer = null;
+    // }
+
+    /**
+     * @{inheritDoc
+     */
+    public String getName() {
+        return "h2sql";
+    }
+
+    /**
+     * @{inheritDoc
+     */
+    public void store(Item item) {
+        if (item.getState() == null || item.getState() instanceof UnDefType) {
+            logger.debug("H2SQL: State of {} [{}] is {}. Store aborted", item.getName(), item.getClass()
+                    .getSimpleName(), item.getState());
+            return;
+        }
+
+        // Connect to H2SQL server if we're not already connected
+        if (!isConnected()) {
+            connectToDatabase();
+        }
```

Comment:
isn't there probably a leaner method to check the connection than isConnected(). Is connected() seem to take a timeout of 5000ms which will potentially slows down the whole persistenceService implementation (not only this Service)